pkg/ride/functions_proto.go

```diff
@@ -310,6 +310,23 @@ func intFromState(env Environment, args ...rideType) (rideType, error) {
 	return rideInt(entry.Value), nil
 }
 
+func intFromSelfState(env Environment, args ...rideType) (rideType, error) {
+	k, err := extractKey(args)
+	if err != nil {
+		return rideUnit{}, nil
+	}
+	a, ok := env.this().(rideAddress)
+	if !ok {
+		return rideUnit{}, nil
```

Comment:
let's return a specifc error like "failed to convert argument to ride address" or something like this

Reply:
Same reason, it have to return `Unit` on every error.